### PR TITLE
feat: export libraries

### PR DESCRIPTION
**Pull request by 0xShaito**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ Solidity Exporter Action automates the process of extracting TypeScript interfac
 | out          | path to the `out` folder containing the compiled contracts                                                                               | out            |                       |
 | interfaces   | path to the interfaces directory                                                                                                         | src/interfaces |                       |
 | contracts    | path to the contracts directory                                                                                                          | src/contracts  |                       |
+| libraries    | path to the libraries directory                                                                                                          | src/libraries  |                       |
 | export_type  | `interface` for exporting only the interfaces and their ABIs, `contracts` for exporting the contracts, interfaces and their ABIs as well | interfaces     | interfaces, contracts |
```
Comment (0xDiscotech) [inline README.md]: Add libraries (if the path was provided) as well here

**Review by 0xDiscotech — APPROVED** [review]
Amazing job!

**Review by turtlemoji — APPROVED** [review]

File: src/createPackage.ts [hunk 2]
```diff
@@ -38,8 +38,13 @@ export const createPackage = (
 
   // Copy the contracts and libraries only if the export type is contracts
   if (exportType === ExportType.CONTRACTS) {
+    // Break the execution in case contractsDir is not defined
+    if(contractsDir == '') {
+      console.error('contractsDir is not defined while exportType is CONTRACTS');
+      throw new Error('contractsDir is not defined while exportType is CONTRACTS');
```
Comment (0xJabberwock) [inline src/createPackage.ts]: What if a package is meant to contain only libraries?

**Review by 0xJabberwock — APPROVED** [review]

**Review by turtlemoji — DISMISSED** [review]

File: src/createPackage.ts [hunk 3]
```diff
@@ -35,8 +38,11 @@ export const createPackage = (
   // Copy the interfaces and their ABIs
   copySolidityFiles(outDir, interfacesDir, destinationDir);
 
-  // Copy the contracts only if the export type is contracts
-  if (exportType === ExportType.CONTRACTS) copySolidityFiles(outDir, contractsDir, destinationDir);
+  // Copy the contracts and libraries only if the export type is all
+  if (exportType === ExportType.ALL) {
+    (contractsDir != '') && copySolidityFiles(outDir, contractsDir, destinationDir);
+    (librariesDir != '') && copySolidityFiles(outDir, librariesDir, destinationDir);
```
Comment (0xGorilla) [inline src/createPackage.ts]: https://discord.com/channels/837716795213217882/1226888069517545482/1271181739552800968

**Review by 0xGorilla — APPROVED** [review]

**Pull request merged**